cmd/validate/validateNavigation.go

```diff
@@ -28,3 +43,81 @@ func validateNavigation(cwd string) {
 		}
 	}
 }
+
+func parseJSONIDs(data map[string]interface{}, file string) {
+	duplicateCounter = make(map[string]int)
+	idValue, ok := data["id"]
+	if idMap, ok := idValue.(string); ok {
+		duplicateCounter[idMap] = 1
+	}
+	navItems, ok := data["navItems"].([]interface{})
+	if ok {
+		for i := 0; i < len(navItems); i++ {
+			navItem, ok := navItems[i].(map[string]interface{})
+			if ok {
+				value, ok := navItem["id"]
+				if ok {
+					if id, ok := value.(string); ok {
+						if _, exists := duplicateCounter[id]; exists {
+							panic(fmt.Sprintf("The id %s in %s is not valid because it is duplicated\n", id, file))
+						} else {
+							duplicateCounter[id] = 1
+						}
+						// FOR DEBUGGING PURPOSES
+						// fmt.Println("----")
+						// fmt.Println(id)
+					}
+				}
+				checkForNestedItems(file, navItem)
+			} 
+		}
+	}
+}
+
+func checkForNestedItems(file string, navItem map[string]interface{}) {
```

Comment:
The `checkForNestedItems` seems to be almost identical to the `parseJSONIDs`. The only difference are the nested items key. Can we merge these two functions into one?

Also, the two branches within the if statement are very similar, I think these can be extracted into a function.